### DISCUSSION TITLE
why @babel/plugin-proposal-class-properties and can't set loose true

### DISCUSSION THREAD

**Discussion by su6838354** [mobxjs/mobx#3658] [General]
office document：make(Auto)Observable only supports properties that are already defined. Make sure your [compiler configuration is correct](https://mobx.js.org/installation.html#use-spec-compliant-transpilation-for-class-properties), or as work-around, that a value is assigned to all properties before using make(Auto)Observable. Without correct configuration, fields that are declared but not initialized (like in class X { y; }) will not be picked up correctly.

try use @babel/plugin-proposal-class-properties and set loose true
```
class Bork {
  static a = "foo";
  static b;

  x = "bar";
  y;
}
```
transform to 
```
var Bork = function Bork() {
  babelHelpers.classCallCheck(this, Bork);
  this.x = "bar";
  this.y = void 0;
};

Bork.a = "foo";
Bork.b = void 0;
```

the class property y is still initialized with undefined？
of course ts will has promble，but the @babel/plugin-proposal-class-properties and set loose true has no promble

**Comment by urugator** [comment]
Because our implementation assumes `false` (standard ES behavior) and we don't have to worry about `true`.
